### PR TITLE
refactor: removing walletpublicid inside our app

### PR DESCRIPTION
**Pull request by nicolasburtey**

### CODE REVIEW

File: src/app/wallets/get-csv-for-wallets.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-import { accountPath } from "@services/ledger/accounts"
+import { walletPath } from "@services/ledger/accounts"
 
 export const getCSVForWallets = async (
   walletIds: WalletId[],
 ): Promise<string | ApplicationError> => {
   const csv = new CSVAccountExport()
   for (const walletId of walletIds) {
-    await csv.addAccount({ account: accountPath(walletId) })
+    await csv.addAccount({ account: walletPath(walletId) })
```
Comment (bodymindarts) [inline src/app/wallets/get-csv-for-wallets.ts:9]: This seems to still be on the wrong side of the core / ledger boundary - is that right?
Not saying it needs to be done in this PR just asking to check if we have a shared understanding.

File: src/app/wallets/intraledger-send-payment.ts [hunk 2]
```diff
@@ -157,15 +157,19 @@ const executePaymentViaIntraledger = async ({
...
     walletId,
   })
+
   if (intraledgerLimitCheck instanceof Error) return intraledgerLimitCheck
 
   const recipientUser = await UsersRepository().findByUsername(recipientUsername)
   if (recipientUser instanceof Error) return recipientUser
   if (recipientUser.id === userId) return new SelfPaymentError()
+  // FIXME: selfPayment should be at the walletId level, no longer at the UserId
+  // but this is not a bloquer until we have multiple wallets per account
```
Comment (bodymindarts) [inline src/app/wallets/intraledger-send-payment.ts:167]: Typo in `blocker` and I prefer to have comments above the line that they refer to....

no big deal.

File: src/core/lightning/wallet.ts [hunk 3]
```diff
@@ -35,7 +35,7 @@ export class LightningUserWallet extends OnChainMixin(UserWallet) {
       logger: this.logger,
     })
 
-    return redlock({ path: this.user._id, logger: this.logger }, async () => {
+    return redlock({ path: this.user.id, logger: this.logger }, async () => {
```
Comment (bodymindarts) [inline src/core/lightning/wallet.ts:38]: Will this change be an issue if there is a lock inflight during upgrade?

File: src/domain/ledger/index.ts [hunk 4]
```diff
@@ -31,8 +31,8 @@ export const ExtendedLedgerTransactionType = {
   LnIntraLedger: "ln_on_us",
 } as const
 
-export const toWalletId = (accountId: LiabilitiesAccountId): WalletId | null => {
-  const path = accountId.split(":")
+export const toWalletId = (walletIdPath: LiabilitiesWalletId): WalletId | null => {
```
Comment (bodymindarts) [inline src/domain/ledger/index.ts:34]: I thought we agreed to go with `CoreAccount` and `LiabilitiesAccount` but here its still `LiabilitiesWallet`? Is this pending a follow up PR?

File: src/domain/wallets/index.ts [hunk 5]
```diff
@@ -1,19 +1,21 @@
-import { InvalidPublicWalletId } from "@domain/errors"
+// import { InvalidWalletId } from "@domain/errors"
```
Comment (bodymindarts) [inline src/domain/wallets/index.ts:1]: Delete dead code?

File: src/domain/wallets/index.ts [hunk 6]
```diff
@@ -1,19 +1,21 @@
...
-  if (!walletPublicId.match(WalletPublicIdRegex)) {
-    return new InvalidPublicWalletId(walletPublicId)
-  }
-  return walletPublicId as WalletPublicId
+export const checkedToWalletId = (walletId: string): WalletId | ValidationError => {
+  return walletId as WalletId
+
+  // TODO: uncomment after walletId is using uuid v4
+
+  // if (!walletId.match(WalletIdRegex)) {
```
Comment (bodymindarts) [inline src/domain/wallets/index.ts:17]: Committed dead code?

File: src/migrations/20211209211641-use-walletid-for-ledger.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
+  async up(db) {
+    {
+      const result = await db.collection("users").updateMany({}, [
```
Comment (bodymindarts) [inline src/migrations/20211209211641-use-walletid-for-ledger.ts:4]: I thought this PR should be non-risk ie. without a migration?

File: src/app/wallets/get-csv-for-wallets.ts (hunk 1 above)
Comment (nicolasburtey) [reply src/app/wallets/get-csv-for-wallets.ts:9]: that is correct. `walletPath` should not be used outside ledger (eventually)

File: src/app/wallets/intraledger-send-payment.ts (hunk 2 above)
Comment (nicolasburtey) [reply src/app/wallets/intraledger-send-payment.ts:167]: oh yeah I had used the french version lol

File: src/core/lightning/wallet.ts (hunk 3 above)
Comment (nicolasburtey) [reply src/core/lightning/wallet.ts:38]: `_id` and `.id` are the same value, so no.

File: src/domain/ledger/index.ts (hunk 4 above)
Comment (nicolasburtey) [reply src/domain/ledger/index.ts:34]: yes

File: src/app/wallets/get-csv-for-wallets.ts (hunk 1 above)
Comment (bodymindarts) [reply src/app/wallets/get-csv-for-wallets.ts:9]: Right so probably this use case is not on `app/wallets` but `app/ledger`

File: src/core/lightning/wallet.ts (hunk 3 above)
Comment (bodymindarts) [reply src/core/lightning/wallet.ts:38]: afaik one of them is a string serizliazation and the other is an object

File: src/migrations/20211209211641-use-walletid-for-ledger.ts (hunk 7 above)
Comment (nicolasburtey) [reply src/migrations/20211209211641-use-walletid-for-ledger.ts:4]: this is a migration just to copy `walletPublicId` to `walletId` in the user collection. this is not the risky part. 

the "risky" part is related to the update of the Ledger

**Review by bodymindarts — DISMISSED** [review]
I'm approving to move forward. Don't see any critical issues.

File: src/servers/graphql-old-server.ts [hunk 8]
```diff
@@ -384,7 +361,7 @@ const resolvers = {
         let feeSatAmount: Satoshis | ApplicationError
         if (amount && amount > 0) {
           feeSatAmount = await getNoAmountLightningFee({
-            walletPublicId: wallet.user.walletPublicId,
+            walletId: wallet.user.id,
```
Comment (vindard) [inline src/servers/graphql-old-server.ts:364]: Should this be `wallet.user.walletId` after the migration?

**Review by vindard — COMMENTED** [review]
Just one question about the name of the `walletId` property on the `UserType` object we get back from mongo. Everything else looks ok to approve after that question.

File: src/servers/graphql-old-server.ts (hunk 8 above)
Comment (nicolasburtey) [reply src/servers/graphql-old-server.ts:364]: I think this should be `user.id` here. where are the other place where we still access `user.walletId`? 

there are some placed that I have replaced here: https://github.com/GaloyMoney/galoy/pull/808/commits/e6d1022f74ca432b0711aae63bfe24b85c98af21

**Review by bodymindarts — APPROVED** [review]

**Review by vindard — APPROVED** [review]

File: src/servers/graphql-old-server.ts (hunk 8 above)
Comment (vindard) [reply src/servers/graphql-old-server.ts:364]: Sorry I was looking at the wrong branch locally 🙈  ... approved ✅

**Pull request merged**